src/kvstore/raftex/RaftPart.cpp

```diff
@@ -419,18 +430,21 @@ folly::Future<AppendLogResult> RaftPart::appendLogAsync(ClusterID source,
 
     LogID firstId = 0;
     TermID termId = 0;
+    AppendLogResult res;
     {
         std::lock_guard<std::mutex> g(raftLock_);
-        auto res = canAppendLogs();
-        if (res != AppendLogResult::SUCCEEDED) {
-            LOG(ERROR) << idStr_
-                       << "Cannot append logs, clean the buffer";
-            sendingPromise_.setValue(std::move(res));
-            replicatingLogs_ = false;
-            return res;
+        res = canAppendLogs();
+        if (res == AppendLogResult::SUCCEEDED) {
+            firstId = lastLogId_ + 1;
+            termId = term_;
         }
-        firstId = lastLogId_ + 1;
-        termId = term_;
+    }
+    if (res != AppendLogResult::SUCCEEDED) {
+        LOG(ERROR) << idStr_
+                   << "Cannot append logs, clean the buffer";
+        sendingPromise_.setValue(std::move(res));
```

Comment:
I think we also need to set cachingPromise to res here. WDYT?

Reply:
En...   We could do that,  but the cachingPromise will be set when sending it.